addons/im_livechat/models/im_livechat_channel.py

```diff
@@ -115,14 +115,14 @@ def _get_available_users(self):
     def _get_livechat_mail_channel_vals(self, anonymous_name, operator, user_id=None, country_id=None):
         # partner to add to the mail.channel
         operator_partner_id = operator.partner_id.id
-        channel_partner_to_add = [(4, operator_partner_id)]
+        channel_partner_to_add = [Command.create({'partner_id': operator_partner_id, 'is_pinned': False})]
         visitor_user = False
         if user_id:
             visitor_user = self.env['res.users'].browse(user_id)
             if visitor_user and visitor_user.active:  # valid session user (not public)
-                channel_partner_to_add.append((4, visitor_user.partner_id.id))
+                channel_partner_to_add.append(Command.create({'partner_id': visitor_user.partner_id.id}))
         return {
-            'channel_partner_ids': channel_partner_to_add,
+            'channel_last_seen_partner_ids': channel_partner_to_add,
```

Comment:
Purpose of 14.3 cleaning was to remove most direct write on that field that should be technical. I suppose issue comes from is_pinned = True by default, does that really makes sense ?

If we have to keep this is_pinned = True, I would suggest a post processing at create to clearly handle cases where we do not want to pin by default, isn't it possible ?

Reply:
This is the only occurrence where `is_pinned = True` is really a problem. Maybe also when creating a DM, but it is much less problematic.

In all other cases of creating or joining a channel is_pinned should be True by default, so the current default value leads to the least overrides so I wouldn't change it, especially not in stable.

Post-processing here was considered first, but it leads to a crappier code with more queries and heuristics to find the proper channel.partner. I think it's much better to make an exception here and create with correct value immediately.